Modules/Tests/YosemiteTests/Tools/POS/POSCatalogSyncCoordinatorTests.swift

```diff
@@ -860,4 +860,78 @@ extension POSCatalogSyncCoordinatorTests {
         // Then - sync should proceed (exactly at 30-day boundary is still eligible)
         #expect(mockIncrementalSyncService.startIncrementalSyncCallCount == 1)
     }
+
+    // MARK: - isSyncStale Tests
+
+    @Test func test_isSyncStale_returns_true_when_no_full_sync_performed() async throws {
+        // Given - no full sync date set
+
+        // When
+        let isStale = await sut.isSyncStale(for: sampleSiteID, maxDays: 7)
+
+        // Then
+        #expect(isStale == true)
+    }
+
+    @Test func test_isSyncStale_returns_false_when_full_sync_is_recent() async throws {
+        // Given - last full sync was 3 days ago
+        let threeDaysAgo = Calendar.current.date(byAdding: .day, value: -3, to: Date())!
```

Comment:
nit: can use `try #require` instead of explicitly unwrapping (similarly for other `!` usage in this file)

```suggestion
        let threeDaysAgo = try #require(Calendar.current.date(byAdding: .day, value: -3, to: Date()))
```